lightly/models/modules/masked_vision_transformer_timm.py

```diff
@@ -74,21 +74,58 @@ def forward(
         if self.vit.attn_pool is not None:
             x = self.vit.attn_pool(x)
         elif self.vit.global_pool == "avg":
-            x = x[:, self.vit.num_prefix_tokens :].mean(dim=1)
+            x = x[:, self.vit.num_prefix_tokens:].mean(dim=1)
         elif self.vit.global_pool:
             x = x[:, 0]  # class token
         return x
 
-    def encode(
+    def forward_intermediates(
         self,
         images: Tensor,
         idx_mask: Optional[Tensor] = None,
         idx_keep: Optional[Tensor] = None,
-    ) -> Tensor:
+    ) -> List[Tensor]:
         """Encode input images.
 
+                Args:
+                    images:
+                        Batch of input images.
+                    idx_mask:
+                        Tensor with shape (batch_size, num_tokens_to_mask) where each
+                        entry is an index of the token to mask in the respective batch.
+                        If specified, the indexed tokens are masked with self.mask_token.
+                    idx_keep:
+                        Tensor with shape (batch_size, num_tokens_to_keep) where each
+                        entry is an index of the token to keep in the respective batch.
+                        If specified, only the indexed tokens will be encoded.
+
+                Returns:
+                    Batch of encoded output tokens.
+                """
+        # preprocess images, convert to tokens and add positional embeddings
+        tokens = self.preprocess(images=images, idx_mask=idx_mask, idx_keep=idx_keep)
+        # normalization layer
+        tokens = self.vit.norm_pre(tokens)
+
+        blocks = self.vit.blocks
+        intermediates: List[Tensor] = []
+        for i, blk in enumerate(blocks):
+            tokens = blk(tokens)
+            intermediates.append(tokens)
+
+        return intermediates
```

Comment:
Could you also add this part from TIMM: https://github.com/huggingface/pytorch-image-models/blob/a1996ec0f43dc7fcc053de7048be6a51cff2ee08/timm/models/vision_transformer.py#L731-L736

Because in most cases you want intermediates and the final output.